src/Npgsql/TypeHandlers/NumericHandlers/NumericHandler.cs

```diff
@@ -169,6 +244,8 @@ public override int ValidateAndGetLength(decimal value, NpgsqlParameter? paramet
         public int ValidateAndGetLength(double value, NpgsqlParameter? parameter) => ValidateAndGetLength((decimal)value, parameter);
         /// <inheritdoc />
         public int ValidateAndGetLength(byte value, NpgsqlParameter? parameter)   => ValidateAndGetLength((decimal)value, parameter);
+        /// <inheritdoc />
+        public int ValidateAndGetLength(SqlDecimal value, NpgsqlParameter? parameter) => ValidateAndGetLength((decimal)value, parameter);
```

Comment:
This looks like it will silently truncate the value.

Ideally, we'd fully support writing SqlDecimal just like you've added support for reading it. If you'd rather not, it seems better to throw NotSupportedException.